app/models/conversation.rb

```diff
@@ -86,7 +86,7 @@ def subscribers(user)
   end
 
   def receive(user, person)
-    cnv = Conversation.find_or_create_by!(self.attributes)
+    cnv = Conversation.create_with(self.attributes).find_or_create_by!(self.attributes[:guid])
```

Comment:
Already tried that. In that case we create a second conversation in the cukes which makes them fail: 
![](https://cloud.githubusercontent.com/assets/3798614/6266466/76a49256-b83a-11e4-95db-12427d3d0dee.png)